src/partial.ts

```diff
@@ -0,0 +1,76 @@
+import { purry } from './purry';
+
+type RemovePrefix<
+  T extends ReadonlyArray<unknown>,
+  U extends ReadonlyArray<unknown>,
+> = T extends [...U, ...infer V] ? V : never;
```

Comment:
You need to add `readonly` here to make this generic enough for all cases because mutable arrays extend readonly arrays, but not vice-versa.

```suggestion
> = T extends readonly [...U, ...infer V] ? V : never;
```